calculator_test.go

```diff
@@ -24,31 +24,34 @@ func TestSubtract(t *testing.T) {
 }
 
 type TestCase struct {
-	A    float64
-	B    float64
-	want float64
+	method string
+	got    interface{}
+	want   float64
 }
 
-func TestMultiply(t *testing.T) {
+func TestCalculator(t *testing.T) {
 	t.Parallel()
 	testCases := []TestCase{
 		{
-			A:    2,
-			B:    2,
-			want: 4,
+			method: "Add",
+			got:    calculator.Add(2, 2),
```

Comment:
Yes, this is a perfectly reasonable way of doing it. (In this case the type of your `got` field would be `float64`, not `interface{}`, because we know exactly what concrete type it will be.)

But there's a neat alternative to this. As you know, functions are values in Go, so we can put the _function itself_ as part of the test case. (Not a _function call_, as you have here, which calls the function with the specified inputs and assigns the _result_ to the struct field, but a _function value_, that is to say, `calculator.Add`, for example.)

The tricky bit is working out what type the function field should be. The answer is it's `func something...`, where the exact type is determined by the function signature. For your Add, Subtract, and Multiply functions, it would be `func(float64, float64) float64`. In other words, a function's type is its signature minus the names.

How do we call such a function value? It's easy! Just like any other function. For example, if the struct field is called `f`, then we call it like this:

```go
tc.f(2, 2)
```

Reply:
Your alternative was my initial solution but the compiler complained about my function signature `func(a, b float64) float64` . So I unleashed the power of interfaces :D